### PR TITLE
[URP 2D] Renderer Feature Support

### PR DESCRIPTION
**Pull request by rainsing**
### Purpose of this PR
Renderer Feature Support for the 2D Renderer.

---
### Testing status
Manual testing + new GFX test 

---
### Comments to reviewers
N/A

### CODE REVIEW

File: com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md [hunk 1]
```diff
@@ -32,3 +32,9 @@ All layers captured for use in the supplied texture will be drawn from the very
 
 ### Downsampling Method
 Downsampling will reduce the texture resolution used by _CameraSortingLayerTexture_. The options are: __None__, __2x Bilinear__, __4x Box__, __4x Bilinear__
+
+## Renderer Features
+
+The 2D Renderer supports Renderer Features. The setup for the features are called before any of the 2D built-in passes are queued.
```
Comment (Sam-Unity) [inline com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md:38]: Give a brief explanation about what 'Render Features' are and maybe what they are for in this line.

**Review by Sam-Unity — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md (hunk 1 above)
Comment (rainsing) [reply com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md:38]: I think this is addressed by the line below (link to renderer feature doc page)?

Comment (Sam-Unity) [reply com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md:38]: The user would need to know if the features are relevant to them before clicking the link to it.

Comment (Sam-Unity) [reply com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md:38]: Suggest this line just to have the same title as the linked page:
"The 2D Renderer supports [URP Renderer Features](urp-renderer-features). The setup for the features are called before any of the 2D built-in passes are queued. Refer to the **URP Renderer Features** documentation for more information."

Comment (rainsing) [reply com.unity.render-pipelines.universal/Documentation~/2DRendererData_overview.md]: Did what you suggested in the latest commit.

**Review by UnityPaul — APPROVED** [review]

**Review by Sam-Unity — APPROVED** [review]
Approving the doc changes to 2DRendererData_Overview.md

**Review by unity-cchu — APPROVED** [review]

**Comment by BoonKiangTan** [comment]
Fix verified on branch:

Version: 2021.2.0a3.484
Revision: trunk 453023b6d02d
Built: Tue, 19 Jan 2021 22:35:46 GMT
Branch: 2d/renderer-feature (bdbbe256ec)

- Verified "Render Objects" Renderer Features in "060_2D_Referer_Feature" test scene work as intended
- No issues with UI and no errors or warnings in console

**Review by BoonKiangTan — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs [hunk 2]
```diff
@@ -176,6 +176,11 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
 
             ConfigureCameraTarget(colorTargetHandle.Identifier(), depthTargetHandle.Identifier());
 
+            // Add passes from Renderer Features.
+            isCameraColorTargetValid = true;    // This is to make it possible to call ScriptableRenderer.cameraColorTarget in the custom passes.
+            AddRenderPasses(ref renderingData);
+            isCameraColorTargetValid = false;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:182]: I'd advise against allowing to call cameraColorTarget in AddRenderPasses if you ever have some logic to decide to draw to staging texture or backbuffer. That's why we prevent that in our renderer. We can't know before we figure pass dependencies.

**Review by phi-lira — APPROVED** [review]
Before merging, please run Nightly test as you are adding test image to make sure the new test passes on all platforms (we don't run all in ABV)
 
 Also you have to add test images to consoles.

**Review by hdb-unity — APPROVED** [review]
**Note:** did not test the changes. Did not confirm CI results.

Checked that developer is aware to add test project changes to UniversalGraphicsTest_2D.

Ideally please confirm before merging the PR:
- No new failures in CI test run
- Revert changes to UniversalGraphicsTest (unless they are required)

**Pull request merged**